index.bs

```diff
@@ -4462,7 +4469,7 @@ partial dictionary MLOpSupportLimits {
     1. If |hiddenSize| * 6 is not a [=valid dimension=], then [=exception/throw=] a {{TypeError}}.
         <details class=note>
           <summary>Why |hiddenSize| * 6 ?</summary>
-          Some underlying platforms operate on a single bias tensor which is a concatenation of {{MLGruCellOptions/bias}} and {{MLGruCellOptions/recurrentBias}}. Therefore, 3 * |hiddenSize| + 3 * |hiddenSize| must also be a [=valid dimension=].
+          Some underlying platforms operate on a single bias tensor which is a concatenation of {{MLGruCellOptions/bias}} and {{MLGruCellOptions/recurrentBias}}. Therefore, 3 * |hiddenSize| + 3 * |hiddenSize| also needs to be a [=valid dimension=].
```

Comment:
Was `must` inappropriate here? It is a spec requirement that `3 * |hiddenSize| + 3 * |hiddenSize|` is a valid dimension.

Reply:
For RFC2119 conformance, the magic word "MUST" in specs is an implementation requirement. That is, this is something that the browser implementation of WebNN MUST do. This informative note is not stating an implementation requirement, so it's not appropriate. So thinking through who the note is for...

(1) a potential script author; but then it would need to be phrased as "you MUST pass X where X * 6 is a valid dimensions"; but then it is *not* an implementation requirement, so can't use the magic word MUST, or
(2) an implementer trying to write a conforming implementation; but then it would need to be phrased as "if X * 6 is not a valid dimension, then the implementation MUST throw an exception" - which *is* an implementation requirement... but already stated in the algorithm, because algorithms in modern web specs are implicitly all MUST requirements.

Ultimately, this note isn't really _either_ those those; it's explaining the "why" of the text to a potential implementer (or proxy, like a reviewer), and IMHO that's not a normative requirement. So... the best thing to do is just avoid the word.

Reply:
🤔 Increasingly I'm wondering whether this "RFC2119" is more detrimental to reader clarity than helpful o_o... 🤷